spark-lint/src/main/kotlin/com/adevinta/spark/lint/UnknownStringResourceAnnotationsDetector.kt

```diff
@@ -0,0 +1,148 @@
+                !in VALID_ANNOTATION_NAMES -> context.reportUnknownAnnotationNameIssue(attribute)
+                COLOR_ANNOTATION_NAME -> checkColorAnnotationValue(context, attribute)
+                TYPOGRAPHY_ANNOTATION_NAME -> checkTypographyAnnotationValue(context, attribute)
+            }
+        }
+    }
+
+    private fun checkColorAnnotationValue(context: XmlContext, node: Node) {
+        if (node.nodeValue !in VALID_COLOR_ANNOTATION_VALUES) {
+            Incident(context)
+                .issue(UNKNOWN_ANNOTATION_VALUE_ISSUE)
+                .at(node)
+                .message(
+                    "${node.nodeValue} is not a valid value for annotation \"color\"" +
+                        "\nPossible values are $VALID_COLOR_ANNOTATION_VALUES",
+                )
+                .report()
+        }
+    }
+
+    private fun checkTypographyAnnotationValue(context: XmlContext, node: Node) {
+        if (node.nodeValue !in VALID_TYPOGRAPHY_ANNOTATION_VALUES) {
+            Incident(context)
+                .issue(UNKNOWN_ANNOTATION_VALUE_ISSUE)
+                .at(node)
+                .message(
+                    "${node.nodeValue} is not a valid value for annotation \"typography\"" +
+                        "\nPossible values are $VALID_TYPOGRAPHY_ANNOTATION_VALUES",
+                )
+                .report()
+        }
+    }
+
+    private fun XmlContext.reportUnknownAnnotationNameIssue(node: Node) =
+        Incident(this)
+            .issue(UNKNOWN_ANNOTATION_NAME_ISSUE)
+            .at(node)
+            .message("Unknown annotation name")
+            .report()
+
+    internal companion object {
+        private const val TAG_ANNOTATION = "annotation"
+        private const val COLOR_ANNOTATION_NAME = "color"
+        private const val TYPOGRAPHY_ANNOTATION_NAME = "typography"
+
+        private val VALID_ANNOTATION_NAMES = listOf(COLOR_ANNOTATION_NAME, TYPOGRAPHY_ANNOTATION_NAME)
+        private val VALID_COLOR_ANNOTATION_VALUES = listOf(
+            "main",
+            "support",
+            "success",
+            "alert",
+            "error",
+            "info",
+            "neutral",
+            "accent",
+        )
+        private val VALID_TYPOGRAPHY_ANNOTATION_VALUES = listOf(
+            "display1",
+            "display2",
+            "display3",
+            "headline1",
+            "headline2",
+            "subhead",
+            "large",
+            "body1",
+            "body2",
+            "caption",
+            "small",
+            "callout",
+        )
+
+        val UNKNOWN_ANNOTATION_NAME_ISSUE = Issue.create(
+            id = "UnknownStringResourceAnnotationsDetector",
+            briefDescription = "Unknown annotation name",
+            explanation = "This annotation name is not supported and won't be parsed. " +
+                "Supported names are $VALID_ANNOTATION_NAMES",
+            category = CORRECTNESS,
+            priority = 8,
+            severity = ERROR,
```

Comment:
Feel free to let me know if you think the priority or severity should be changed!